pkg/logger/logger.go

```diff
@@ -0,0 +1,67 @@
+package logger
+
+import (
+	"os"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap/zapcore"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var logLevelMapping = map[string]int{
+	"devel":   0,
+	"default": 1, // default one when not set log-mode
+	"prod":    2,
+}
+
+// in each controller, to use different log level.
+func LogWithLevel(logger logr.Logger, level string) logr.Logger {
+	if len(level) == 0 {
+		level = "default"
+	}
+	verbosityLevel, ok := logLevelMapping[level]
+	if !ok {
+		verbosityLevel = 1 // fallback to info level
+	}
+	return logger.V(verbosityLevel)
+}
+
+// in DSC component, to use different mode for logging, e.g. development, production
+// when not set mode it falls to "default" which is used by startup main.go.
+func ConfigLoggers(mode string) logr.Logger {
```

Comment:
I'm wondering, with the planned log from context, will it change?

Reply:
if you meant the proposal from last meeting, it might be a totally different way of how we handle loggers.
we might not need this function on the component level at all. but i am not sure what that solution will be.

Reply:
i think here is only to move similar functionalities into one package, so we can better organize later

Reply:
Yes, about that. Sure, fine to revisit later.